packages/@aws-cdk/aws-iam/lib/managed-policy.ts

```diff
@@ -252,6 +255,28 @@ export class ManagedPolicy extends Resource implements IManagedPolicy {
       resourceName: this.physicalName,
     });
 
+    const self = this;
+
+    class GrantPrincipal implements IPrincipal {
+      public readonly grantPrincipal: IPrincipal = this;
+
+      public get assumeRoleAction(): string {
+        throw new Error('ManagedPolicy has no principals.');
+      }
+      public get policyFragment(): PrincipalPolicyFragment {
```

Comment:
Same comment as above.

Reply:
And a couple places below but I won't add the comment elsewhere to avoid redundancy.